### ISSUE TITLE
(Sprint 2) Update dependencies

### ISSUE THREAD

**Issue by WillBAnders** [ufssd/ufssd-website#20]
Update all dependencies to the latest versions (`package.json` & `package-lock.json`).

## Requirements

 - `npm outdated` should indicate all dependencies are up to date. If any dependencies can't be updated due to bugs or other issues, we'll address those on a case-by-case basis.

**Comment by angel1254mc** [comment]
### Output from running `npm outdated`
![image](https://github.com/ufssd/ufssd-website/assets/59091763/067ab144-679b-4801-b8a8-0e92f82eea42)

[Question]: Do we want to update to Next 14? Most breaking changes don't affect us, but the [minimum node version got bumped](https://nextjs.org/docs/pages/building-your-application/upgrading/version-14#v14-summary) up to '>=18.17.0'. We'd probably need to bump up the node version our actions workflows as well. I don't really have a preference so let me know what you think is best 😊.

**Comment by WillBAnders** [comment]
I'm inclined to update to Next.js 14 myself - our project is pretty low risk, so might as well jump on that for the learning experience and it's easier to do so now than delay until later. However, it's good practice to do these larger framework updates separately to minimize risk (also noticing there's a [v14.0.1 pre-release](https://github.com/vercel/next.js/releases/tag/v14.0.1-canary.0), so no harm in waiting a little bit for that to settle).

Regarding Node version, our actions are configured for `18.x` - you can see it's running the latest `18.18.2` in [this job](https://github.com/ufssd/ufssd-website/actions/runs/6670383400/job/18130132615), so there's no conflict there. I am, however, noticing that we have `@types/node` `v20` instead - this is not correct as it corresponds to Node `v20`, and thus one of these will need to change. Ignore this for now; I'll open a separate ticket to update to Node `v20` as well.

TLDR: We should update to Next.js `v14` and Node `v20`, but will do both of these separately.

**Comment by angel1254mc** [comment]
Got it sounds good 😎👍
